test/integration/imagechange_buildtrigger_test.go

```diff
@@ -477,12 +479,21 @@ func TestMultipleImageChangeBuildTriggers(t *testing.T) {
 		if err != nil {
 			t.Fatalf("Couldn't create Image: %v", err)
 		}
+
+		var newBuild *buildapi.Build
+		var event watchapi.Event
 		// wait for initial build event from the creation of the imagerepo
-		event := <-watch.ResultChan()
+		for {
+			event = <-watch.ResultChan()
+			newBuild = event.Object.(*buildapi.Build)
```

Comment:
At first I would expect `watch.ResultChan()` to return objects of the same type, but I learned that's a false assumption when `oc start-build` started to see `*unversioned.Status` (and not a panic because of the `_, ok` pattern).

If this is all over the place, I'd say have potential panics in all of those places.